var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -40,185 +65,307 @@ class Timemory(CMakePackage):
+        description='Enable LIKWID CPU marker API support (perfmon)',
+    )
+    variant(
+        'likwid_nvmon',
+        default=False,
+        description='Enable LIKWID GPU marker API support (nvmon)',
+    )
     variant('caliper', default=False, description='Enable Caliper support')
-    variant('dyninst', default=False,
-            description='Build dynamic instrumentation tools')
+    variant('dyninst', default=False, description='Build dynamic instrumentation tools')
     variant('examples', default=False, description='Build/install examples')
-    variant('gperftools', default=False,
-            description='Enable gperftools support')
-    variant('kokkos_tools', default=False,
-            description=('Build generic kokkos-tools libraries, e.g. '
-                         'kp_timemory, kp_timemory_filter'))
-    variant('kokkos_build_config', default=False,
-            description=('Build pre-configured (i.e. dedicated) kokkos-tools '
-                         'libraries, e.g. kp_timemory_cpu_flops'))
-    variant('cuda_arch', default='auto', description='CUDA architecture name',
-            values=('auto', 'kepler', 'tesla', 'maxwell', 'pascal',
-                    'volta', 'turing'), multi=False)
-    variant('cpu_target', default='auto',
-            description=('Build for specific cpu architecture (specify '
-                         'cpu-model)'))
-    variant('use_arch', default=False,
-            description=('Build all of timemory w/ cpu_target architecture '
-                         'flags (default: roofline toolkit only)'))
-    variant('tls_model', default='global-dynamic',
-            description='Thread-local static model', multi=False,
-            values=('global-dynamic', 'local-dynamic', 'initial-exec',
-                    'local-exec'))
-    variant('lto', default=False,
-            description='Build w/ link-time optimization')
-    variant('statistics', default=True,
-            description=('Build components w/ support for statistics '
-                         '(min/max/stddev)'))
-    variant('extra_optimizations', default=True,
-            description='Build timemory with extra optimization flags')
-    variant('cxxstd', default='14', description='C++ language standard',
-            values=('14', '17', '20'), multi=False)
-    variant('mpip_library', default=False,
-            description='Build stand-alone timemory-mpip GOTCHA library')
+    variant('gperftools', default=False, description='Enable gperftools support')
+    variant(
+        'kokkos_tools',
+        default=False,
+        description=(
+            'Build generic kokkos-tools libraries, e.g. '
+            'kp_timemory, kp_timemory_filter'
+        ),
+    )
+    variant(
+        'kokkos_build_config',
+        default=False,
+        description=(
+            'Build pre-configured (i.e. dedicated) kokkos-tools '
+            'libraries, e.g. kp_timemory_cpu_flops'
+        ),
+    )
+    variant(
+        'cuda_arch',
```

Comment:
Any reason not to use spack's builtin `cuda_arch` from CudaPackage and map that to your equivalent cmake argument? Notice that cmake 3.18 has https://cmake.org/cmake/help/v3.21/prop_tgt/CUDA_ARCHITECTURES.html which maps even easier to what spack provides. When you extend CudaPackage you get all the conflicts so that people don't get build errors after building all dependencies with a version of gcc that's unsupported by cuda.

Reply:
@haampie I do map these arguments to `CUDA_ARCHITECTURES` internally, but only when the cmake version is > 3.18 but since it supports cmake 3.15-3.17, I have to keep consistency via these arguments.